### PR TITLE
Redo simplex orientation

### PR DESCRIPTION
**Pull request by drwells**
Possibly related to something Timo mentioned on slack today.

To get P3 elements working we need to consistently orient all lines and faces in a 3D simplex triangulation. AFAIK this requires using the same table for all faces in `standard_vs_true_line_orientation()` since the ordering of the faces is somewhat arbitrary (and what may be face 1 on one cell could be face 2 on another). Fixing this table seems to make all line orientations consistent (we get unique support points for line dofs).

I'm still stuck on tetrahedral faces. Faces in orientations 3 and 5 give the wrong answers w.r.t. permutations of face vertices. This is also wrong on master. Here's some sample output:

```
cell 0.2
  vertices = 0, 3, 4, 1
 face = 7 orientation = 1
                  vertices = 0, 3, 4
       vertices via permut = 0, 3, 4
    face quadrature points = 0 0 0, -1 -1 0, 0 0 1
 face = 6 orientation = 1
                  vertices = 3, 0, 1
       vertices via permut = 3, 0, 1
    face quadrature points = -1 -1 0, 0 0 0, 1 -1 0
 face = 0 orientation = 3
                  vertices = 1, 0, 4
       vertices via permut = 4, 1, 0
     computed orientation = 5
    face quadrature points = 0 0 1, 1 -1 0, 0 0 0
 face = 8 orientation = 1
                  vertices = 4, 3, 1
       vertices via permut = 4, 3, 1
    face quadrature points = 0 0 1, -1 -1 0, 1 -1 0
```

for whatever reason permutations 3 and 5 should really be 5 and 3. What saves us on master is that these orientations are, AFAICT, only achievable when tetrahedra have negative volumes but this is no longer the case with the "make lines consistently oriented" renumbering.

### CODE REVIEW

**Comment by tjhei** [comment]
I am not familiar with the simplex details. Let me know when you want me to run my test with your changes.

File: include/deal.II/grid/connectivity.h [hunk 1]
```diff
@@ -196,7 +196,7 @@ namespace internal
         if (d == 1)
           {
             static const dealii::ndarray<unsigned int, 3, 2> table = {
-              {{{0, 1}}, {{1, 2}}, {{2, 0}}}};
+              {{{0, 1}}, {{1, 2}}, {{0, 2}}}};
```
Comment (bangerth) [inline include/deal.II/grid/connectivity.h:199]: This is unrelated to what you're doing, but this kind of thing should really be sitting in `ReferenceCell` as well.

**Review by bangerth — COMMENTED** [review]
Ugh, not a fun patch to review. Not sure I'm up for it, but I'm also not sure whether you're ready for it either?

The way I tested these sorts of orientation issues is to take a global polynomial and either project or interpolate it onto the finite element space on these kinds of meshes. The result needs to be the exact polynomial I started with, and you can check that using `integrate_difference()`. But if you get an orientation wrong, where two adjacent cells disagree where on the common edge multiple DoFs are located, then you get cells where the error is not zero (for both projection and interpolation), and that helps with visual debugging.

I believe that there are still tests of this kind in the test suite, though I don't recall their names.

**Comment by drwells** [comment]
This won't be ready for review until we figure out what is going on with orientations 3 and 5 with tetrahedral faces - that's also wrong on `master`, though I still don't think its possible to presently generate such a face.

Another thing that will help a lot will be adding documentation on orientation - that will explain the changes (and where all these tables come from).

**Comment by peterrum** [comment]
@drwells 

> Possibly related to something Timo mentioned on slack today.

What do you mean by this comment?

**Comment by drwells** [comment]
It has been 6 months and I don't remember!

**Comment by drwells** [comment]
This isn't relevant now that we found the underlying bugs - see the other PR.

**Pull request closed without merging**